issue_drafts/issue_5.md

```diff
@@ -0,0 +1,173 @@
+If so, congratulations! The pipeline is complete.
+
+## Changing and rebuilding pipeline targets
+
+Now, with a complete pipeline, we can play around a bit!
+Let's run some experiments to test Snakemake's limits and better understand when Snakemake does or does not rebuild a part of a pipeline.
+First things first: let's delete all output files and rebuild the pipeline from scratch.
+```
+snakemake -c1 3_plot/out/doy_plot.png --delete-all-output
+snakemake -c1 3_plot/out/doy_plot.png
+```
+> NOTE: -c1 is short for --cores 1
+
+Now let's see what happens when we add another lake to the plot.
+From among all the site IDs of the csvs that are outputs of `unzip_sb_data`, choose one site and add it to the inputs of `combine_site_files`.
+For instance, you could add "2_process/out/doy_86444267.csv" as a third output of `combine_site_files`.
+Now try a dry run to see what would happen if we build the pipeline again.
+```
+snakemake -n
+```
+> NOTE: -n is short for --dry-run
+
+You'll probably see the following message:
+> Building DAG of jobs...
+> The input used to generate one or several output files has changed:
+>     To inspect which output files have changes, run 'snakemake --list-input-changes'.
+>     To trigger a re-run, use 'snakemake -R $(snakemake --list-input-changes)'.
+> Nothing to be done (all requested files are present and up to date).
+
+We know that Snakemake won't rebuild any pipeline targets because it says "Nothing to be done".
+This is important to notice - when the Snakefile is changed, Snakemake doesn't rebuild everything affected by that change by default.
+However, Snakemake did detect the change to inputs that we made, and offers a way to re-run the pipeline with the changes.
+First, let's inspect which output files have changes, as it suggests.
+```
+snakemake --list-input-changes
+```
+One file appears: `2_process/out/combined_doy.csv`. 
+That makes sense - the inputs for that file changed.
+Let's try re-running the pipeline in the way the Snakemake warning message suggests.
+First, let's do a dry run.
+```
+snakemake -n -R $(snakemake --list-input-changes)
+```
+> NOTE: The `-R` flag is short for --forcerun. The flag forces a file to be re-created, even if Snakemake would otherwise not remake it. All dependent (downstream) files are also re-created, but the files that the target depends on (upstream) are not remade.
+
+You can see that, if we execute this command without the `-n` flag, we'll create the averaged temperatures for the new lake in the file `2_process/out/doy_86444267.csv`, and re-create `2_process/out/combined_doy.csv` and `3_plot/out/doy_plot.png`.
+This entails re-running four rules: `calc_doy_means`, `combine_site_files`, `plot_doy_mean`, and `all` (rule `all` doesn't actually create any files).
+Notice that only the new lake's averaged file gets created; `doy_120020150.csv` and `doy_107072210` remain untouched because adding a new lake has no affect on those target files.
```

Comment:
```suggestion
Notice that only the new lake's averaged file gets created; `doy_120020150.csv` and `doy_107072210.csv` remain untouched because adding a new lake has no affect on those target files.
```
```suggestion
Notice that only the new lake's averaged file gets created; `doy_120020150.csv` and `doy_107072210` remain untouched because adding a new lake has no affect on those target files.
```